diagnosis/collectlogs.sh

```diff
@@ -167,28 +179,57 @@ then
     TENANT_ID=$(sudo jq -r '.serviceManagementEndpoint' /etc/kubernetes/azurestackcloud.json | cut -d / -f4)
 fi
 
-for cid in $(docker ps -a -q --no-trunc)
-do
-    cns=$(docker inspect --format='{{ index .Config.Labels "io.kubernetes.pod.namespace" }}' ${cid})
-    
-    # if NAMESPACES not set, then collect everything
-    if [ -z "${NAMESPACES}" ] || (echo $NAMESPACES | grep -qw $cns);
-    then
-        # Ignore the Pause container
-        if docker inspect --format='{{ .Config.Image }}' ${cid} | grep -q -v pause;
+if systemctl list-units | grep -q docker.service; then
```

Comment:
an improvement would be to check if the docker service is running before invoking commands